src/test/java/com/android/volley/toolbox/DiskBasedAsyncCacheTest.java

```diff
@@ -0,0 +1,449 @@
+                        } catch (ExecutionException e) {
+                            // do nothing
+                        } catch (InterruptedException e) {
+                            // do nothing
+                        }
+                        assertThat(listCachedFiles(), is(emptyArray()));
+                    }
+                });
+    }
+
+    @Test
+    public void testReinitialize() throws ExecutionException, InterruptedException {
+        Cache.Entry entry = randomData(1023);
+        putEntry("key", entry).get();
+
+        AsyncCache copy = new DiskBasedAsyncCache(temporaryFolder.getRoot(), MAX_SIZE);
+        copy.initialize(emptyCallback);
+
+        final CompletableFuture<Cache.Entry> getEntry = new CompletableFuture<>();
+        copy.get(
+                "key",
+                new AsyncCache.OnGetCompleteCallback() {
+                    @Override
+                    public void onGetComplete(@Nullable Cache.Entry entry) {
+                        getEntry.complete(entry);
+                    }
+                });
+
+        assertThatEntriesAreEqual(getEntry.get(), entry);
+    }
+
+    @Test
+    public void testInvalidate() throws ExecutionException, InterruptedException {
+        Cache.Entry entry = randomData(32);
+        entry.softTtl = 8765432L;
+        entry.ttl = 9876543L;
+        putEntry("key", entry).get();
+
+        invalidateEntry("key", false).get();
+        entry.softTtl = 0; // expired
+        assertThatEntriesAreEqual(getEntry("key").get(), entry);
+    }
+
+    @Test
+    public void testInvalidateFullExpire() throws ExecutionException, InterruptedException {
+        Cache.Entry entry = randomData(32);
+        entry.softTtl = 8765432L;
+        entry.ttl = 9876543L;
+        putEntry("key", entry).get();
+
+        invalidateEntry("key", true).get();
+        entry.softTtl = 0; // expired
+        entry.ttl = 0; // expired
+        assertThatEntriesAreEqual(getEntry("key").get(), entry);
+    }
+
+    @Test
+    public void testManyResponseHeaders() throws ExecutionException, InterruptedException {
+        Cache.Entry entry = new Cache.Entry();
+        entry.data = new byte[0];
+        entry.responseHeaders = new HashMap<>();
+        for (int i = 0; i < 0xFFFF; i++) {
+            entry.responseHeaders.put(Integer.toString(i), "");
+        }
+        putEntry("key", entry).get();
+    }
+
+    @Test
+    public void initializeIfRootDirectoryDeleted() throws ExecutionException, InterruptedException {
+        temporaryFolder.delete();
+
+        Cache.Entry entry = randomData(101);
+        putEntry("key1", entry).get();
+
+        assertThat(getEntry("key1").get(), is(nullValue()));
+
+        // confirm that we can now store entries
+        putEntry("key2", entry).get();
+        assertThatEntriesAreEqual(getEntry("key2").get(), entry);
+    }
+
+    /* Test helpers */
+
+    /** Puts entry into the cache, and returns a CompletableFuture, to let us know when it's done */
```

Comment:
nit: over the line limit
super nit: Add a period on the end (same for #getEntry)